### PR TITLE
RFQ cleanup

### PR DESCRIPTION
**Pull request by ffranr**
This PR was recently reviewed, approved, and merged: https://github.com/lightninglabs/taproot-assets/pull/835

I noticed that some of the review comments could also apply to other parts of the RFQ code that were not changed in the above PR.

This PR improves the RFQ code following the review comments from https://github.com/lightninglabs/taproot-assets/pull/835.

### CODE REVIEW

File: rfqmsg/buy_accept_test.go [hunk 1]
```diff
@@ -40,3 +41,59 @@ func TestAcceptShortChannelId(t *testing.T) {
...
+	testCases := []struct {
+		testName string
+
+		id       ID
+		askPrice lnwire.MilliSatoshi
+		expiry   uint64
+		sig      [64]byte
+	}{
+		{
+			testName: "all fields populated with basic values",
```
Comment (GeorgeTsagk) [inline rfqmsg/buy_accept_test.go:68]: maybe some more cases? testing failure

Comment (ffranr) [reply rfqmsg/buy_accept_test.go:68]: I've added an "empty fields" test case.

I'm not sure what you mean by "testing failure" in this case. The field types will ensure that the encoding/decoding will work.

**Review by GeorgeTsagk — APPROVED** [review]
lgtm!

File: rfqmsg/buy_accept_test.go [hunk 2]
```diff
@@ -40,3 +41,66 @@ func TestAcceptShortChannelId(t *testing.T) {
...
+			askPrice: 1000,
+			expiry:   42000,
+			sig:      signature,
+		},
+		{
+			testName: "empty fields",
+			id:       [32]byte{},
+			askPrice: 0,
+			expiry:   0,
+			sig:      [64]byte{},
```
Comment (jharveyb) [inline rfqmsg/buy_accept_test.go:79]: I imagine invalid sigs will be rejected at some later (not the decoder) point?

**Review by jharveyb — APPROVED** [review]
Looks good!

**Pull request merged**